pkg/queue/queue.go

```diff
@@ -0,0 +1,180 @@
+/*
+	Copyright 2022 Loophole Labs
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		   http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package queue
+
+import (
+	"github.com/loopholelabs/frisbee/pkg/packet"
+	"github.com/pkg/errors"
+	"runtime"
+	"sync/atomic"
+	"unsafe"
+)
+
+var (
+	Closed = errors.New("queue is closed")
+)
+
+func round(value uint64) uint64 {
```

Comment:
nit: this file in particular isnt very readable. We should add comments to speed up future code reviews as well as any work that touches these files